addons/mail/models/mail_guest.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import pytz
+import uuid
+from odoo.tools import consteq
+from odoo import fields, models, _
+from odoo.addons.base.models.res_partner import _tz_get
+
+class MailGuest(models.Model):
+    _name = 'mail.guest'
+    _description = "Guest"
+
+    access_token = fields.Char(
+        copy=False,
+        default=lambda self: str(uuid.uuid4()),
+        groups='base.group_system',
+        readonly=True,
+        required=True,
+    )
+    channel_ids = fields.Many2many(comodel_name='mail.channel')
+    country_id = fields.Many2one(string="Country", comodel_name='res.country')
+    lang_id = fields.Many2one(string="Language", comodel_name='res.lang')
+    name = fields.Char(required=True)
+    timezone = fields.Selection(selection=_tz_get)
+
+    def format_guests(self):
+        formatted_guests = [{
+            'id': guest.id,
+            'name': guest.name,
+        } for guest in self]
+
+        return formatted_guests
+
+    def _get_guest_from_request(self, request):
+        # Retrieve required data from cookies.
+        guest_id = request.httprequest.cookies.get('mail.guest_id')
+        guest_access_token = request.httprequest.cookies.get('mail.guest_access_token')
```

Comment:
do you really need two cookies? isn't the token enough to retrieve an existing guest?

Reply:
Isn't it less safe? I don't think psql will do a consteq equivalent when doing the search for it?

Also that would mean we need to index it on top of indexing the id, just to avoid one extra cookie that contains something as small as an id? I don't mind the change (if it's safe at least), I'm just wondering.

Reply:
Good point. We could use hash of tokens as we do in `res.users.apikeys._check_credentials` but it's more sensitive.